src/rust/engine/fs/src/store.rs

```diff
@@ -279,13 +276,30 @@ impl Store {
         expanded_digests
       })
       .and_then(move |digests| {
+        if Store::upload_is_faster_than_checking_whether_to_upload(digests.clone()) {
+          return Ok((
+            digests
+              .keys()
+              .map(|key| key.clone())
+              .collect::<HashSet<Digest>>(),
+            digests,
+          ));
+        }
+        remote
+          .list_missing_digests(digests.keys())
+          .map(|fitered_digests| (fitered_digests, digests))
+      })
+      .and_then(move |(fitered_digests, digest_entry_types)| {
         future::join_all(
-          digests
+          fitered_digests
             .into_iter()
-            .map(move |(digest, entry_type)| {
-              let remote = remote.clone();
+            .map(move |digest| {
+              let entry_type = digest_entry_types.get(&digest).unwrap();
+              let remote = remote2.clone();
```

Comment:
It's not clear that this clone is necessary... it doesn't look like remote2 is used anywhere else in this closure, so I think you can just use `remote2` directly in `load_bytes_with`?

Reply:
I tried this but it didn't work. This could be totally made up nonsense, but the way I saw it was that you can't move the same variable implicitly more than once...? Hence the clone...